src/main/java/mate/fruitshop/service/FruitService.java

```diff
@@ -0,0 +1,46 @@
+package mate.fruitshop.service;
+
+import static mate.fruitshop.dao.FruitTransactionDaoCsv.FRUIT;
+import static mate.fruitshop.dao.FruitTransactionDaoCsv.QUANTITY;
+import static mate.fruitshop.service.strategy.TransactionHandlerStrategy.TRANSACTIONS_MAP;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import mate.fruitshop.dao.FruitTransactionDao;
+import mate.fruitshop.model.FruitTransaction;
+
+public class FruitService {
+    private FruitTransactionDao dao;
+
+    public FruitService(FruitTransactionDao dao) {
+        this.dao = dao;
+    }
+
+    public Map<String, Integer> calculateFruitsLeft() {
+        List<FruitTransaction> transactions = dao.getAll();
+        Map<String, Integer> fruitQuantityMap = new HashMap<>();
+        int currentFruitQuantity;
+
+        for (FruitTransaction transaction: transactions) {
```

Comment:
```suggestion
        for (FruitTransaction transaction : transactions) {
```
don't forget to autoformat your files (Ctrl Alt L)